File: examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/db/dao/CredentialOfferDao.java
package me.connect.sdk.java.sample.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import me.connect.sdk.java.sample.db.entity.CredentialOffer;

@Dao
public interface CredentialOfferDao {
    @Query("SELECT * FROM credentialoffer")
    LiveData<List<CredentialOffer>> getAll();

    @Query("SELECT * FROM credentialoffer WHERE id = :id")
    CredentialOffer getById(int id);

    @Insert
    void insertAll(CredentialOffer... connections);

    @Query("SELECT EXISTS(SELECT * FROM credentialoffer WHERE (claim_id = :claimId AND pwDid = :pwDid))")
    boolean checkOfferExists(String claimId, String pwDid);

    @Query("SELECT * FROM credentialoffer WHERE (claim_id = :claimId AND pwDid = :pwDid)")
    CredentialOffer getByPwDidAndClaimId(String claimId, String pwDid);

    @Update
    void update(CredentialOffer connection);

    @Delete
    void delete(CredentialOffer connection);
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/msdk/java/samplekt/db/dao/CredentialOfferDao.kt
package msdk.java.samplekt.db.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import msdk.java.samplekt.db.entity.CredentialOffer

@Dao
interface CredentialOfferDao {
    @Query("SELECT * FROM credentialoffer")
    fun getAll(): LiveData<List<CredentialOffer>>

    @Query("SELECT * FROM credentialoffer WHERE id = :id")
    suspend fun getById(id: Int): CredentialOffer

    @Insert
    suspend fun insertAll(vararg connections: CredentialOffer)

    @Query("SELECT EXISTS(SELECT * FROM credentialoffer WHERE (claim_id = :claimId))")
    fun checkOfferExists(claimId: String?): Boolean

    @Query("SELECT * FROM credentialoffer WHERE (claim_id = :claimId AND pwDid = :pwDid)")
    fun getByPwDidAndClaimId(
        claimId: String?,
        pwDid: String?
    ): CredentialOffer?

    @Update
    suspend fun update(connection: CredentialOffer)

    @Delete
    suspend fun delete(connection: CredentialOffer)
}